### DISCUSSION TITLE
Skipping query when parameters are not present

### DISCUSSION THREAD

**Discussion by AlecRobertson** [urql-graphql/urql#2813] [Q&A]
I have a query that takes in params that are required for it to work. I have set the pause parameter in the variables for the query to be true when these params are not present. But the query complains the params might be undefined. How can I work around this?

**Comment by nandorojo** [comment]
A typical strategy is to do this:

```ts
function useUser(id: string | null) {
  return useQuery({ query, variables: { id: id ?? '' }, pause: !id })
}
```

As an aside, I made `id` `string | null` to ensure that you always pass _something_:

```ts
useUser(userId || null)
```

If you just made the argument optional (ie `useUser(id?: string)`), then you might forget to pass an ID when it's consumed.

Finally, I recommend setting `strictNullChecks: true` in your `tsconfig.json`'s `compilerOptions`.
